client/go/outline/client.go

```diff
@@ -33,11 +33,30 @@ type Client struct {
 	transport.PacketListener
 }
 
+// NewClientResult represents the result of [NewClientAndReturnError].
+type NewClientResult struct {
```

Comment:
Why the wrapping struct? Is that needed for Java?

Reply:
It's for returning  the strongly typed `PlatformError`, which can be read in Java.

If we return `(Client, error)`, the `error` will be thrown as a weakly typed `Exception`, so we lose the details (such as `ErrorCode`).

Reply:
But couldn't you return `(Client, platerrors.PlatformError)` here?

Reply:
Unfortunately no, according to [gobind](https://pkg.go.dev/golang.org/x/mobile@v0.0.0-20240905004112-7c4916698cc9/cmd/gobind#hdr-Type_restrictions):

> Any function type all of whose parameters and results have supported types. Functions must return either no results, one result, or two results where the type of the second is the built-in 'error' type.

Reply:
Ah good to know. Is that worth adding in a comment maybe for future readers?

Reply:
added comments.